locator/tablets.cc

```diff
@@ -109,28 +109,91 @@ tablet_replica get_leaving_replica(const tablet_info& tinfo, const tablet_transi
 
 const tablet_map& tablet_metadata::get_tablet_map(table_id id) const {
     try {
-        return _tablets.at(id);
+        return *_tablets.at(id);
     } catch (const std::out_of_range&) {
         throw std::runtime_error(format("Tablet map not found for table {}", id));
     }
 }
 
-tablet_map& tablet_metadata::get_tablet_map(table_id id) {
-    return const_cast<tablet_map&>(
-        const_cast<const tablet_metadata*>(this)->get_tablet_map(id));
+void tablet_metadata::mutate_tablet_map(table_id id, noncopyable_function<void(tablet_map&)> func) {
```

Comment:
Since tablet_metadata is part of token metadata, its updates should be already isolated by the means of mutate_token_metadata(), and work on a temporary copy before it's published to all shards.
Do we really need mutate_tablet_map?

Reply:
Strictly speaking yes, but I still wanted to enforce immutability on the API level.